squidDrone/src/propulsion/motor_builder.cpp

```diff
@@ -0,0 +1,45 @@
+#include "motor_builder.hpp"
+
+#include <set>
+
+#include "stm32g4xx_hal_tim.h"
+
+#ifndef UNIT_TEST
+#include "letodar_2204.hpp"
+#include "little_bee_20_a.hpp"
+#else
+#include "letodar_2204_mock.hpp"
+#include "little_bee_20_a_mock.hpp"
+#endif
+namespace propulsion {
+
+auto MotorBuilder::Create(propulsion::PropulsionHardwareConfig& motor_config) noexcept -> std::unique_ptr<Motor> {
+  std::unique_ptr<Motor> motor_pointer = nullptr;
+  if (MotorConfigIsValid(motor_config)) {
+    auto correct_esc = GetCorrectEsc(motor_config);
+    switch (motor_config.motor_type) {
+      case types::MotorType::LETODAR_2204:
+        motor_pointer = std::make_unique<LeTodar2204>(std::move(correct_esc));
+        break;
+    }
+  } 
+  return motor_pointer;
+}
+
+auto MotorBuilder::MotorConfigIsValid(propulsion::PropulsionHardwareConfig& motor_config) noexcept -> const bool {
+  const std::set<std::uint32_t> possible_channels{TIM_CHANNEL_1, TIM_CHANNEL_2, TIM_CHANNEL_3, TIM_CHANNEL_4, TIM_CHANNEL_5, TIM_CHANNEL_6};
+  const bool channel_is_in_possible_channels = possible_channels.find(motor_config.channel) != possible_channels.end();
+  return (channel_is_in_possible_channels && motor_config.timer != nullptr);
+}
+
+auto MotorBuilder::GetCorrectEsc(propulsion::PropulsionHardwareConfig& config) noexcept -> std::unique_ptr<Esc> {
+  std::unique_ptr<Esc> esc;
```

Comment:
Why no nullptr here?
What happens if config.esc_type is not LITTLE_BEE_20_A. How do you check later if this is valid?

Reply:
Fixed in [this commit](https://github.com/elheck/Squiddrone/pull/230/commits/e04970a97abe893523fc4550125cd271e221a661).